### ISSUE TITLE
Can't SSH into server

### ISSUE THREAD

**Issue by Vilux** [drduh/YubiKey-Guide#148]
I've gotten my public key from my Yubikey 5 with `ssh-add -L`, created a remote server and added the key to the `authorized_keys` file and have tried logging into it with `ssh user@159.89.127.161` with the device plugged in, however I am getting a 
`sign_and_send_pubkey: signing failed: agent refused operation`
`user@159.89.127.161: Permission denied (publickey)` error
Why is the server refusing my key?

**Comment by AlexanderBartash** [comment]
Hi. Can you please check if you have any keys in `~/.ssh`. Those might be interfering with your GPG keys.

**Comment by Vilux** [comment]
I have id_rsa.pub and id_rsa_yubikey.pub which both have the public key

**Comment by AlexanderBartash** [comment]
I have only `known_hosts` in `~/.ssh`. It might be worth to try to temporarily move those public keys out of `~/.ssh` and see if it helps.

**Comment by Vilux** [comment]
Just tried, didn't help. Does the `ssh_config` file need to be modified?

**Comment by AlexanderBartash** [comment]
Actually did you do this:

Add this to `~/.bashrc`
```
export GPG_TTY="$(tty)"
export SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket)
gpgconf --launch gpg-agent
```

Then add this to ~/.gnupg/gpg-agent.conf
See https://github.com/drduh/config/blob/master/gpg-agent.conf
See https://github.com/dainnilsson/scripts/blob/master/base-install/gpg.sh
See https://github.com/drduh/YubiKey-Guide#steps-for-older-distributions
```
enable-ssh-support
default-cache-ttl 60
max-cache-ttl 120
```

After you edit ~/.gnupg/gpg-agent.conf need to restart agent
```
gpg-connect-agent killagent /bye && gpg-connect-agent /bye
```

**Comment by Vilux** [comment]
I already had all of that stuff set. Same error after restarting gpg-agent :(

**Comment by AlexanderBartash** [comment]
You may have seen this already but here https://lists.gnupg.org/pipermail/gnupg-users/2018-August/060885.html A guy was having similar issue when he locked his yubikey by too many PIN failures.

**Comment by AlexanderBartash** [comment]
Anyway. I am out of ideas. Google reveals a few results. I suppose you've tired it :) https://www.google.com/search?q=gpg+ssh+signing+failed%3A+agent+refused+operation

**Comment by Vilux** [comment]
Thanks for your help. It looks like running 
`echo UPDATESTARTUPTTY | gpg-connect-agent` allowed me to ssh in

**Comment by Wheest** [comment]
I recently moved to a new machine, and I had a similar issue after setting up my Yubikey, and trying to run `ssh`.

```
$ ssh remote
sign_and_send_pubkey: signing failed for RSA "cardno:XXXXX" from agent: agent refused operation
```

However, the source/solution to my issue was the following:

In my `.gnupg/gpg-agent.conf` I had the following line, copied from my old configuration:

`pinentry-program /usr/bin/pinentry-gtk-2`

However, I did not have this program installed on the new machine.  I commented out the line, restarted my GPG agent with `gpgconf --kill gpg-agent`, and was able to connect (using the native pinentry-program of my desktop environment).  I could have also installed the pinentry-gtk-2 package, or set the program as `pinentry-program /usr/bin/pinentry-curses`.
